include/seqan3/alphabet/structure/structured_aa.hpp

```diff
@@ -0,0 +1,170 @@
+ * // W W B
+ *
+ * // modify via structured bindings and references:
+ * auto & [ seq_l, structure_l ] = l;
+ * seq_l = aa27::V;
+ * std::cout << to_char(l) << '\n';
+ * // V
+ *
+ * ~~~~~~~~~~~~~~~
+ *
+ * This seqan3::cartesian_composition itself fulfills seqan3::alphabet_concept.
+ */
+
+template <typename sequence_alphabet_t = aa27, typename structure_alphabet_t = dssp9>
+requires alphabet_concept<sequence_alphabet_t> && alphabet_concept<structure_alphabet_t>
+struct structured_aa :
+    public cartesian_composition<structured_aa<sequence_alphabet_t, structure_alphabet_t>,
+                                 sequence_alphabet_t, structure_alphabet_t>
+{
+    //!\brief First template parameter as member type.
+    using sequence_alphabet_type = sequence_alphabet_t;
+    //!\brief Second template parameter as member type.
+    using structure_alphabet_type = structure_alphabet_t;
+
+    //!\brief Equals the char_type of sequence_alphabet_type.
+    using char_type = underlying_char_t<sequence_alphabet_type>;
+    //!\brief Equals the char_type of the structure_alphabet_type.
+    using str_char_type = underlying_char_t<structure_alphabet_type>;
+
+    /*!\name Write functions
+     * \{
+     */
+    //!\brief Directly assign the sequence character.
+    constexpr structured_aa & operator=(sequence_alphabet_type const l) noexcept
+    {
+        get<0>(*this) = l;
+        return *this;
+    }
+
+    //!\brief Directly assign the structure character.
+    constexpr structured_aa & operator=(structure_alphabet_type const l) noexcept
+    {
+        get<1>(*this) = l;
+        return *this;
+    }
+
+    //!\brief Assign from a nucleotide character. This modifies the internal sequence letter.
+    constexpr structured_aa & assign_char(char_type const c)
+    {
+        seqan3::assign_char(get<0>(*this), c);
+        return *this;
+    }
+
+    //!\brief Assign from a structure character. This modifies the internal structure letter.
+    constexpr structured_aa & assign_structure(str_char_type const c)
+    {
+        seqan3::assign_char(get<1>(*this), c);
+        return *this;
+    }
+    //!\}
+
+    /*!\name Read functions
+     * \{
+     */
+    //!\brief Return the structure character. This reads the internal structure letter.
+    constexpr str_char_type to_structure() const noexcept
+    {
+        return seqan3::to_char(get<1>(*this));
+    }
+
+    //!\brief Return a character. This reads the internal sequence letter.
+    constexpr char_type to_char() const noexcept
+    {
+        return seqan3::to_char(get<0>(*this));
+    }
+    //!\}
+};
+
+//!\brief Type deduction guide enables usage of structured_aa without specifying template args.
+//!\relates structured_aa
+template <typename sequence_alphabet_type, typename structure_alphabet_type>
+structured_aa(sequence_alphabet_type &&, structure_alphabet_type &&)
+    -> structured_aa<std::decay_t<sequence_alphabet_type>, std::decay_t<structure_alphabet_type>>;
```

Comment:
but there is no `structured_aa(a, b)` constructor, is there? Have you tested this?

Reply:
Yes, I tested constructing with the { } brackets (see the structured_aa_type_deduce test). 
This deduction is necessary to initialise without template arguments: 
`structured_aa my_letter{aa27::C, dssp9::G};`

I see that it does not work with ( ) brackets, but do we require this?

Reply:
> I see that it does not work with ( ) brackets, but do we require this?

No, that's ok.